api/src/main/java/org/geysermc/geyser/api/pack/ResourcePack.java

```diff
@@ -59,6 +62,25 @@ public interface ResourcePack {
     @NonNull
     String contentKey();
 
+    /**
+     * Sets the content key of the resource pack. Lack of a content key can be represented by an empty string.
+     */
+    void contentKey(@NonNull String contentKey);
```

Comment:
```suggestion
    /**
     * Sets the content key of the resource pack. Lack of a content key can be represented by an empty string.
     */
    void contentKey(@Nullable String contentKey);
```
for consistency with subpacks